### PR TITLE
dry-validation 1.0

### PR DESCRIPTION
**Pull request by mensfeld**
close https://github.com/karafka/waterdrop/issues/86
close https://github.com/karafka/waterdrop/issues/85

this PR is ready for review unless @solnic decides to change more things :D

### CODE REVIEW

File: spec/lib/water_drop/schemas/config_spec.rb [hunk 1]
```diff
@@ -150,6 +150,7 @@
       before { config[:kafka][:seed_brokers] = %w[#$^&* ^&*()] }
 
       it { expect(schema.call(config)).to be_failure }
+      it { p schema.call(config).errors }
```
Comment (madejejej) [inline spec/lib/water_drop/schemas/config_spec.rb]: oops :)

**Comment by solnic** [comment]
There are small tweaks I could do but 0 time for it. I'll send another PR later this month.

File: waterdrop.gemspec [hunk 2]
```diff
@@ -19,8 +19,8 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'delivery_boy', '~> 0.2'
   spec.add_dependency 'dry-configurable', '~> 0.8'
   spec.add_dependency 'dry-monitor', '~> 0.3'
-  spec.add_dependency 'dry-validation', '~> 0.11'
-  spec.add_dependency 'ruby-kafka', '>= 0.7.1'
+  spec.add_dependency 'dry-validation', '~> 1.0.0.rc1'
```
Comment (solnic) [inline waterdrop.gemspec]: I *think* if you set it to `~> 1.0.0` already it will work with RCs too.

Comment (mensfeld) [reply waterdrop.gemspec]: Nope:
```Bundler could not find compatible versions for gem "dry-validation":
  In Gemfile:
    dry-validation

    waterdrop was resolved to 1.3.0, which depends on
      dry-validation (~> 1.0.0)

Could not find gem 'dry-validation (~> 1.0.0)', which is required by gem 'waterdrop', in any of the relevant sources:
  https://github.com/dry-rb/dry-validation.git (at master@d748280)

```

**Pull request merged**